### PR TITLE
Handle new identifier structure

### PR DESCRIPTION
**Pull request by BerniVarga**
This PR adjust the RW for the new identifier structures. 

The changes imply:
- at write time: instead of looking at concept nodes uuid, the UPPIdentifier's value is checked. If the concept node is not yet loaded: create the concept node as Thing{uuid}, a UPPIdentifier{value=uuid} node, and the IDENTIFIES relationship between them
- the delete query is also updated to delete all the annotation relationships for v1 concepts, and only MENTIONS relationships for v2 concepts

### CODE REVIEW

File: annotations/cypher.go [hunk 1]
```diff
@@ -59,6 +59,7 @@ func (s service) Read(contentUUID string) (thing interface{}, found bool, err er
 	//TODO shouldn't return Provenances if none of the scores, agentRole or atTime are set
 	statementTemplate := `
 					MATCH (c:Thing{uuid:{contentUUID}})-[rel{platformVersion:{platformVersion}}]->(cc:Thing)
+					MATCH (cc:Thing)
```
Comment (nwrigh) [inline annotations/cypher.go]: Why is this needed?

**Comment by tosan88** [comment]
:+1:

**Comment by nwrigh** [comment]
:+1:
